testapi.pm

```diff
@@ -1359,22 +1359,30 @@ sub assert_shutdown {
     $timeout //= 60;
     bmwqemu::log_call(timeout => $timeout);
     $timeout = bmwqemu::scale_timeout($timeout);
-    while ($timeout >= 0) {
-        my $is_shutdown = query_isotovideo('backend_is_shutdown') || 0;
-        if ($is_shutdown < 0) {
-            bmwqemu::diag("Backend does not implement is_shutdown - just sleeping");
-            sleep($timeout);
-        }
-        # -1 counts too
-        if ($is_shutdown) {
-            $autotest::current_test->take_screenshot('ok');
-            return;
+    if (check_var('BACKEND', 'svirt')) {
```

Comment:
@mnowaksuse looks like you are destroying the object oriented approach here. Try to avoid if/else by using polymorphy